src/controller/python/BUILD.gn

```diff
@@ -63,7 +63,6 @@ shared_library("ChipDeviceCtrl") {
       "chip/internal/ChipThreadWork.h",
       "chip/internal/CommissionerImpl.cpp",
       "chip/logging/LoggingRedirect.cpp",
```

Comment:
It does make sense to have a method called at python startup that is not tied to a specific class like "DeviceController" or similar. This is what StackInit was supposed to do.

Reply:
The unfortunate aspect of stack initialization in Matter is that you need to pick a side as part of initialization - controller (`DeviceControllerFactory::Init`), or server (`Server::Init`). You cannot complete stack initialization without calling one of those these two functions.

And only after calling those, can you then actually spool up the Matter thread.

Reply:
We _could_ tease out the common bits of stack initialization shared by the two. It's very small (memory init, and platform stack init), but that could then be shared by the Server version of Python initialization.

We can do this as a follow-up, since even in the code today, the Server initialization side of Python (see `ServerInit.cpp`) completely duplicates `StackInit`, which meant that `StackInit` was not actually being used in that capacity.

Reply:
I think we should clearly separate what can and cannot be done in terms of init independent from others.

In an ideal world, dependencies would be clear, but they do not seem so right now. However for a path forward at least, we should not just put everything in one basket. Specifically I disagree that MemoryInit should be part of the device controller factory or server init - that should stay where it was.

Reply:
I've refactored the common stack initialization bits into CommonStackInit().